### PR TITLE
Update to jasmine 3.4 & fix resulting spec failures

### PR DESCRIPTION
**Pull request by raphinesse**
Replaces #74 together with #79.

### CODE REVIEW

File: spec/superspawn.spec.js [hunk 1]
```diff
@@ -31,9 +30,12 @@ describe('spawn method', function () {
         failSpy = jasmine.createSpy('fail'); /* eslint no-unused-vars : 0 */
     });
 
-    it('Test 001 : should return a promise', function () {
-        expect(Q.isPromise(superspawn.spawn(LS))).toBe(true);
-        expect(Q.isPromise(superspawn.spawn('invalid_command'))).toBe(true);
+    it('should resolve on success', () => {
+        return expectAsync(superspawn.spawn(LS)).toBeResolved();
```
Comment (brodybits) [inline spec/superspawn.spec.js]: Possibly off-topic: I think it should be possible to use a very dumb command such as `echo` that is exactly the same between Windows cmd, ps, and *nix *sh (including Windows "*nix*" *sh). This should probably be done in a separate PR, someday. Not a blocker.

Comment (raphinesse) [reply spec/superspawn.spec.js]: Would be nice. Although I hope we will sunset `superspawn` before that :wink:

**Comment by raphinesse** [comment]
@brodybits thanks for the fast response. Just waiting for AppVeyor CI to succeed.

**Comment by brodybits** [comment]
Yeah, AppVeyor now seems to be slow on Apache Cordova. I generally setup Travis CI & AppVeyor CI on my forks whenever possible, for another sanity check. And +1 for sunsetting `superspawn`, as it would really help finish removing `q`. And I had one more idea, maybe worth $0.02:

In case of improvements that are very closely related, I would personally try to propose them together in a single PR, to be merged in a merge commit, for the sake of clarity. (Also avoid the chance of others doing parallel investigations) Just an idea.

I wish GitHub would have some better support for related PRs. Also for people to suggest changes to multiple lines.

**Comment by erisu** [comment]
@raphinesse I am wondering if the fix you applied for the failing spec can also be applied in iOS.

I pined a few repo's Jasmine dependency to `~3.1.0` to prevent `3.2.0` which broke some tests. I didn't have time to investigate at that time but wondering now if it is a similar issue.

https://github.com/apache/cordova-ios/issues/393

**Comment by raphinesse** [comment]
@erisu yeah, I think I see the issue there. It's basically unhandled promises 95% of the time. The rest are oddities related to the combination of Jasmine and Q.

Will try to post a fix the following days.
